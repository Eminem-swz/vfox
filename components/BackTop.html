<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>BackTop 返回顶部 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.133cf44e.css">
    <link rel="modulepreload" href="/vfox/assets/Home.999026eb.js">
    <link rel="modulepreload" href="/vfox/assets/app.3c8a5e7e.js">
    <link rel="modulepreload" href="/vfox/assets/components_BackTop.md.570d2a8e.lean.js">
    
    <meta name="twitter:title" content="BackTop 返回顶部 | Vfox for Vue3">
  <meta property="og:title" content="BackTop 返回顶部 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-3eed0030><div class="sidebar-button" data-v-3eed0030><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-3eed0030 data-v-d91ad9e0><!----> Vfox for Vue3</a><div class="flex-grow" data-v-3eed0030></div><div class="nav" data-v-3eed0030><nav class="nav-links" data-v-3eed0030 data-v-65a69246><!--[--><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/" data-v-a00c0156>指南 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/design/" data-v-a00c0156>设计 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item active" href="/vfox/components/Button" data-v-a00c0156>组件 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/playground/" target="_blank" data-v-a00c0156>Playground <!----></a></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-1314d532><nav class="nav-links nav" data-v-1314d532 data-v-65a69246><!--[--><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/" data-v-a00c0156>指南 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/design/" data-v-a00c0156>设计 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item active" href="/vfox/components/Button" data-v-a00c0156>组件 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/playground/" target="_blank" data-v-a00c0156>Playground <!----></a></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-1314d532><!--[--><li class="sidebar-link"><p class="sidebar-link-item">基础</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Button">Button</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Icon">Icon</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Image">Image</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Drawer">Drawer</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Modal">Modal</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dropdown">Dropdown</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">反馈</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dialog">Dialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Notify">Notify</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SwipeCell">SwipeCell</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">数据输入</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Form">Form</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Input">Input</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Radio">Radio</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Checkbox">Checkbox</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Slider">Slider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Switch">Switch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stepper">Stepper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Calendar">Calendar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SearchBar">SearchBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NumberKeyboard">NumberKeyboard</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImageUploader">ImageUploader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Range">Range</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Rate">Rate</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">导航</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/components/BackTop">BackTop</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#import">Import</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#props">Props</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#events">Events</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#slots">Slots</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#点击区域（-default）">点击区域（#default）</a><!----></li></ul></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NavBar">NavBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopMenu">PopMenu</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tab">Tab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SideTab">SideTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabBar">TabBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabView">TabView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Sticky">Sticky</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollTab">ScrollTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/IndexView">IndexView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Fixed">Fixed</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Pagination">Pagination</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">展示</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Layout">Layout</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cell">Cell</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Group">Group</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Collapse">Collapse</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Empty">Empty</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Skeleton">Skeleton</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Swiper">Swiper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImagePreview">ImagePreview</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/FlatList">FlatList</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NoticeBar">NoticeBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Divider">Divider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Badge">Badge</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Progress">Progress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tag">Tag</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Order">Order</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountDown">CountDown</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Steps">Steps</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Price">Price</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Avatar">Avatar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Timeline">Timeline</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stopwatch">Stopwatch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Result">Result</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/LoadMore">LoadMore</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TimeAgo">TimeAgo</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountUp">CountUp</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CircleProgress">CircleProgress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActivityIndicator">ActivityIndicator</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">其他</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Copy">Copy</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-33252f1f><div class="container" data-v-33252f1f><!--[--><!--[--><!----><!--]--><!--[--><div class="demo"><iframe style="" src="/vfox/demo/#/BackTop" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-33252f1f><div><h1 id="backtop-返回顶部" tabindex="-1">BackTop 返回顶部 <a class="header-anchor" href="#backtop-返回顶部" aria-hidden="true">#</a></h1><h2 id="import" tabindex="-1">Import <a class="header-anchor" href="#import" aria-hidden="true">#</a></h2><div class="language-JavaScript"><pre><code><span class="token keyword">import</span> <span class="token punctuation">{</span> BackTop <span class="token punctuation">}</span> <span class="token keyword">from</span> <span class="token string">&#39;vfox&#39;</span>
</code></pre></div><p>具体的引入方式可以参考<a href="./../guide/import.html">引入组件</a>。</p><h2 id="props" tabindex="-1">Props <a class="header-anchor" href="#props" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>visible-height</td><td>number</td><td>200</td><td>否</td><td>滚动高度达到此参数值才出现</td></tr><tr><td>animated</td><td>boolean</td><td>true</td><td>否</td><td>是否需要动画</td></tr><tr><td>offset</td><td>number[]</td><td>[0, 0]</td><td>否</td><td>偏移量，格式为 [x, y]</td></tr><tr><td>enable-safe-area-insets</td><td>boolean</td><td>true</td><td>否</td><td>是否开启安全区适配</td></tr></tbody></table><p>注：</p><p>在 iPhoneX 等机型开启 <code>enableSafeAreaInsets</code>，需要在 head 标签中添加 meta 标签，并设置 viewport-fit=cover 值</p><div class="language-HTML"><pre><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>meta</span>
  <span class="token attr-name">name</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>viewport<span class="token punctuation">&quot;</span></span>
  <span class="token attr-name">content</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>width=device-width, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0, viewport-fit=cover<span class="token punctuation">&quot;</span></span>
<span class="token punctuation">/&gt;</span></span>
</code></pre></div><h2 id="events" tabindex="-1">Events <a class="header-anchor" href="#events" aria-hidden="true">#</a></h2><table><thead><tr><th>事件</th><th>描述</th><th>回调函数参数</th></tr></thead><tbody><tr><td>click</td><td>回到顶部按钮点击时触发</td><td>e: Event</td></tr></tbody></table><h2 id="slots" tabindex="-1">Slots <a class="header-anchor" href="#slots" aria-hidden="true">#</a></h2><h3 id="点击区域（-default）" tabindex="-1">点击区域（#default） <a class="header-anchor" href="#点击区域（-default）" aria-hidden="true">#</a></h3><div class="language-Vue"><pre><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>fx-back-top</span><span class="token punctuation">&gt;</span></span>
  回顶部
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>fx-back-top</span><span class="token punctuation">&gt;</span></span>
</code></pre></div><h4 id="default-的缺省内容" tabindex="-1">default 的缺省内容 <a class="header-anchor" href="#default-的缺省内容" aria-hidden="true">#</a></h4><p>默认显示 <svg width="20px" height="20px" viewBox="64 64 896 896" focusable="false"><path d="M518.5 360.3a7.95 7.95 0 00-12.9 0l-178 246c-3.8 5.3 0 12.7 6.5 12.7H381c10.2 0 19.9-4.9 25.9-13.2L512 460.4l105.2 145.4c6 8.3 15.6 13.2 25.9 13.2H690c6.5 0 10.3-7.4 6.5-12.7l-178-246z"></path><path d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"></path></svg> 图标。</p></div></div><footer class="page-footer" data-v-33252f1f data-v-c772b490><div class="edit" data-v-c772b490><div class="edit-link" data-v-c772b490 data-v-11add020><!----></div></div><div class="updated" data-v-c772b490><!----></div></footer><div class="next-and-prev-link" data-v-33252f1f data-v-6889d424><div class="container" data-v-6889d424><div class="prev" data-v-6889d424><a class="link" href="/vfox/components/Rate" data-v-6889d424><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-6889d424><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-6889d424>Rate</span></a></div><div class="next" data-v-6889d424><a class="link" href="/vfox/components/NavBar" data-v-6889d424><span class="text" data-v-6889d424>NavBar</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-6889d424><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"components_actionsheet.md\":\"476f176a\",\"components_activityindicator.md\":\"d668f69e\",\"components_avatar.md\":\"1fb2d309\",\"components_backtop.md\":\"570d2a8e\",\"components_badge.md\":\"f391b25b\",\"components_button.md\":\"086e0f62\",\"components_calendar.md\":\"beaa3eac\",\"components_cascader.md\":\"c5dc18d4\",\"components_cell.md\":\"edbf7f64\",\"components_checkbox.md\":\"f7d33cef\",\"components_circleprogress.md\":\"66496fa1\",\"components_collapse.md\":\"284da919\",\"components_copy.md\":\"5212c987\",\"components_countdown.md\":\"fe83298d\",\"components_countup.md\":\"ccdb13be\",\"components_datepicker.md\":\"934e5657\",\"components_dialog.md\":\"ccdef5c2\",\"components_divider.md\":\"7ece7c07\",\"components_drawer.md\":\"21fd08c9\",\"components_dropdown.md\":\"30b4c9ce\",\"components_empty.md\":\"e07c29da\",\"components_fixed.md\":\"061f47b7\",\"components_flatlist.md\":\"76814f20\",\"components_form.md\":\"d413c144\",\"components_group.md\":\"6baec165\",\"components_icon.md\":\"d59f3606\",\"components_image.md\":\"68c26bd4\",\"components_imagepreview.md\":\"b11e5c65\",\"components_imageuploader.md\":\"904497b8\",\"components_indexview.md\":\"5d3e8c1d\",\"components_input.md\":\"bc75ea41\",\"components_layout.md\":\"69f518a0\",\"components_loadmore.md\":\"2a822443\",\"components_modal.md\":\"064a5398\",\"components_navbar.md\":\"41876298\",\"components_noticebar.md\":\"873b3688\",\"components_notify.md\":\"897cc498\",\"components_numberkeyboard.md\":\"7b385b5d\",\"components_order.md\":\"aba3db0d\",\"components_pagination.md\":\"854640cb\",\"components_picker.md\":\"7061144d\",\"components_popdialog.md\":\"3ebb5200\",\"components_popmenu.md\":\"80c76469\",\"components_popover.md\":\"073f47f9\",\"components_price.md\":\"b71cd544\",\"components_progress.md\":\"0205898f\",\"components_radio.md\":\"dd96fb6d\",\"components_range.md\":\"901b3b5b\",\"components_rate.md\":\"10f5e8a4\",\"components_result.md\":\"7e4d3ed7\",\"components_scrolltab.md\":\"1d6d4e98\",\"components_scrollview.md\":\"2eed99b3\",\"components_searchbar.md\":\"de180032\",\"components_sidetab.md\":\"9aa1d275\",\"components_skeleton.md\":\"4b4117b3\",\"components_slider.md\":\"11ff5949\",\"components_stepper.md\":\"1bea34d9\",\"components_steps.md\":\"386be8e2\",\"components_sticky.md\":\"7841bebb\",\"components_stopwatch.md\":\"778d75f9\",\"components_swipecell.md\":\"892ad3f4\",\"components_swiper.md\":\"e44dd496\",\"components_switch.md\":\"a824a4b1\",\"components_tab.md\":\"dcc1a36c\",\"components_tabbar.md\":\"bb03228a\",\"components_tabview.md\":\"e262d6de\",\"components_tag.md\":\"fab14512\",\"components_timeago.md\":\"f5cef35b\",\"components_timeline.md\":\"8fc14816\",\"components_toast.md\":\"0ac178c5\",\"components_index.md\":\"c61ef603\",\"design_color.md\":\"0aa8a1c4\",\"design_index.md\":\"c42c48b2\",\"guide_dark.md\":\"7077e797\",\"guide_import.md\":\"8e8c1efb\",\"guide_locale.md\":\"e45f0078\",\"guide_theme.md\":\"57061e24\",\"index.md\":\"34608736\"}")</script>
    <script type="module" async src="/vfox/assets/app.3c8a5e7e.js"></script>
    
  </body>
</html>