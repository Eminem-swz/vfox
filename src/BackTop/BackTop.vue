<template>
  <div
    v-show="scrollTop >= visibleHeight"
    class="fx-back-top"
    :style="styles"
    @click="onClick"
  >
    <slot>
      <icon icon="UpCircleOutlined" />
    </slot>
  </div>
</template>

<script>
import safeAreaInsets from 'safe-area-insets'
import Icon from '../Icon'
import { addScrollEvent, removeScrollEvent } from '../helpers/events'
import { getScrollDom } from '../helpers/dom'

export default {
  name: 'fx-back-top',
  components: { Icon },
  props: {
    visibleHeight: {
      type: Number,
      default: 200
    },
    // 是否需要动画
    animated: {
      type: Boolean,
      default: true
    },
    // 偏移量，格式为 [x, y]
    offset: {
      type: Array,
      default() {
        return [0, 0]
      }
    },
    // 是否开启安全区
    enableSafeAreaInsets: {
      type: Boolean,
      default: true
    }
  },
  data() {
    return {
      scrollTop: 0,
      duration: 200,
      safeAreaInsetBottom: 0
    }
  },
  computed: {
    styles() {
      return {
        transform: `translate3d(${this.offset[0]}px, ${this.offset[1] -
          this.safeAreaInsetBottom}px, 0px)`
      }
    }
  },
  created() {
    this.scrollTop = getScrollDom().scrollTop

    addScrollEvent(this.onScroll)
  },
  mounted() {
    safeAreaInsets.onChange(this.updateSafeAreaInsets)
    this.updateSafeAreaInsets()
  },
  beforeUnmount() {
    safeAreaInsets.offChange(this.updateSafeAreaInsets)
    removeScrollEvent(this.onScroll)
  },
  methods: {
    updateSafeAreaInsets() {
      this.safeAreaInsetBottom = this.enableSafeAreaInsets
        ? safeAreaInsets.bottom
        : 0
    },

    onScroll(e) {
      this.scrollTop = e.target.documentElement.scrollTop
    },

    onClick(e) {
      this.toTop()

      this.$emit(e.type, e)
    },

    toTop() {
      getScrollDom().scrollTo({
        top: 0,
        behavior: this.animated ? 'smooth' : 'instant'
      })
    }
  }
}
</script>
