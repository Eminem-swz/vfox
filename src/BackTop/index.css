.fx-back-top {
  position: fixed;
  right: 20px;
  bottom: 20px;
  z-index: 1000;
  cursor: pointer;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 40px;
  min-height: 40px;
  font-size: 17px;
  --back-top-color: #262626;
  color: var(--back-top-color);
}

.fx-back-top .fx-icon {
  --size: 40px;
  --color: var(--back-top-color);
  display: block;
}
