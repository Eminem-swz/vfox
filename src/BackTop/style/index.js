import '../../Icon/index.css'
import '../index.css'