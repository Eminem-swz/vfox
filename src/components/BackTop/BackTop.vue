<template>
  <div
    v-show="scrollTop >= visibleHeight"
    :class="[prefix + '-back-top']"
    :style="styles"
    @click="onClick"
  >
    <slot>
      <icon class-name="UpCircleOutlined" />
    </slot>
  </div>
</template>

<script>
import Icon from '../Icon/Icon.vue'
import { SDKKey } from '../../config'
import { pageScrollTo } from '../../apis/Scroll'
import { addScrollEvent, removeScrollEvent } from '../../helpers/events'

export default {
  name: SDKKey + '-back-top',
  components: { Icon },
  props: {
    visibleHeight: {
      type: Number,
      default: 200
    },
    // 是否需要动画
    animated: {
      type: Boolean,
      default: true
    },
    // 偏移量，格式为 [x, y]
    offset: {
      type: Array,
      default() {
        return [0, 0]
      }
    }
  },
  data() {
    return {
      prefix: SDKKey,

      scrollTop: 0,
      duration: 200
    }
  },
  computed: {
    styles() {
      return {
        transform: `translate3d(${this.offset[0]}px, ${this.offset[1]}px, 0px)`
      }
    }
  },
  watch: {},
  created() {
    this.scrollTop = document.documentElement.scrollTop

    addScrollEvent(this.onScroll)
  },
  ready() {},
  mounted() {},
  updated() {},
  attached() {},
  beforeDestroy() {
    removeScrollEvent(this.onScroll)
  },
  methods: {
    onScroll(e) {
      this.scrollTop = e.target.documentElement.scrollTop
    },

    onClick(e) {
      this.toTop()

      this.$emit(e.type, e)
    },

    toTop() {
      pageScrollTo({
        scrollTop: 0,
        duration: this.duration
      })
    }
  }
}
</script>

<style lang="scss">
@import '../component.module.scss';

.#{$prefix}-back-top {
  position: fixed;
  right: 20px;
  bottom: 20px;
  z-index: 1000;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 40px;
  min-height: 40px;
  font-size: 17px;
  --back-top-color: #{$title-color};
  color: var(--back-top-color);

  .#{$prefix}-icon {
    --size: 40px;
    --color: var(--back-top-color);
    display: block;
  }
}
</style>
