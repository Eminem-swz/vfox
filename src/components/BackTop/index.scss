@import '../style/var.scss';

.#{$prefix}-back-top {
  position: fixed;
  right: 20px;
  bottom: 20px;
  z-index: 1000;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 40px;
  min-height: 40px;
  font-size: 17px;
  --back-top-color: #{$title-color};
  color: var(--back-top-color);

  .#{$prefix}-icon {
    --size: 40px;
    --color: var(--back-top-color);
    display: block;
  }
}
