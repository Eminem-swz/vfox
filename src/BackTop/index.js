import BackTop from './BackTop.vue'

BackTop.install = function(Vue) {
  Vue.component(BackTop.name, BackTop)
}

export default BackTop
