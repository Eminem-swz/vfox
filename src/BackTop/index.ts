import { App } from 'vue'
import { SFCWithInstall } from '../utils/types'
import BackTop from './BackTop.vue'

const _BackTop: SFCWithInstall<typeof BackTop> = Object.assign(BackTop, {
  install: function(app: App) {
    app.component(BackTop.name, BackTop)
  }
})

export default _BackTop
