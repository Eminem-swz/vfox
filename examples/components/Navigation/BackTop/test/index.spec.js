/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import BackTop from '@/components/BackTop'

describe('BackTop', () => {
  test('should render default correctly', () => {
    const wrapper = mount(BackTop, {})

    expect(wrapper.html()).toMatchSnapshot()
  })
})
