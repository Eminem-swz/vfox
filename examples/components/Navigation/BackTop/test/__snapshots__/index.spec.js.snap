// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackTop should render default correctly 1`] = `
<div class="fx-back-top" style="transform: translate3d(0px, 0px, 0px); display: none;">
  <fx-icon-stub></fx-icon-stub>
</div>
`;

exports[`BackTop should render prop offset=[0, -50] correctly 1`] = `
<div class="fx-back-top" style="transform: translate3d(0px, -50px, 0px); display: none;">
  <fx-icon-stub></fx-icon-stub>
</div>
`;

exports[`BackTop should render set default slot correctly 1`] = `<div class="fx-back-top" style="transform: translate3d(0px, 0px, 0px); display: none;">BackTop</div>`;
